overlord/patch/patch7.go

```diff
@@ -0,0 +1,45 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package patch
+
+import (
+	"os"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/overlord/state"
+)
+
+func init() {
+	patches[7] = patch7
+}
+
+// patch7:
```

Comment:
I see. How about a patch that only removes duplicated cookies from state? This would be rollback-friendly.

Alternatively,  such code could be run on snampgr startup but it will unfortunately need to read all cookies from the disk first just to know which of the duplicates to remove.

Reply:
Ok, I've removed all the patch code. I've implemented new logic in SyncCookies method (previously named GeneratedCookies).